packages/theme/src/Provider.tsx

```diff
@@ -117,11 +117,13 @@ const useChangeTheme = (): ((themeName: string) => void) => {
 
 const withTheme = <Props extends { theme: ReadTheme }>(
   Component: React.ComponentType<Props>
-): React.ComponentType<Omit<Props, "theme">> => {
-  return (props: Omit<Props, "theme">) => {
-    const { theme } = React.useContext(ThemeContext);
-    return <Component {...(props as Props)} theme={theme} />;
-  };
+) => {
+  return React.forwardRef(
+    (props: Omit<Props, "theme">, ref: React.Ref<any>) => {
```

Comment:
With components that need a forwarded ref, I usually don't wrap them in a `withTheme`, instead I get the theme by using the `useTheme` hook and just export them as is. See the `AudioPlayerWithInterface.tsx` for example. By doing that, the type of the ref can be maintained and is not `any` like it is here. This works as well though, it'll just lose the type

Reply:
Thanks, adjust BottomSheet component